crates/sparrow-logical/src/functions.yml

```diff
@@ -0,0 +1,6 @@
+- name: add
```

Comment:
Thinking it might be nice if this was just a data file like this, rather than the complexity of function registration in the old codebase. I suspect with the new approach, there shouldn't *need* to be much here (possibly just a flag indicating which of these are aggregations).

I'm happy to make this a struct and have a rust file with all of them, but thought I'd prototype it as a file to see if we like this better. Thoughts?

Reply:
The old pattern requires a lot of additional flags -- time domain checks, is_new patterns, rewrites, etc. I don't think it would be _messier_ to have it in a yml vs. registration as rust, so no strong opinion here.

Reply:
I suppose we'd miss some type checking / compilation benefits of doing it in rust, but we could still share shared patterns like `AGGREGATION_IS_NEW`, so :shrug:

Reply:
Yep. My hope is that we can make it simpler, and possibly just have something like `is_aggregation` if needed. If this gets to be too messy, I'd lean towards going back to a big list of structs, but possibly all in one place so we can just see the list of entries. But, one benefit of doing this in YML is it may force us to keep it simpler. I'm happy to make it a struct right now, but maybe it's worth *trying* YML to see if it ends up better than what we had?